tabbycat/draw/generator/pairing.py

```diff
@@ -172,27 +173,24 @@ def winner(self):
         return self.teams[self._winner_index]
 
 
-class BPPairing(BasePairing):
-    """Pairing class for British Parliamentary."""
-
-    sides = ['og', 'oo', 'cg', 'co']
+class PolyPairing(BasePairing):
+    """Pairing class for British Parliamentary and Public Speaking."""
 
-    def __init__(self, teams, bracket, room_rank, flags=[], team_flags={}):
+    def __init__(self, teams, bracket, room_rank, num_sides=4, flags=[], team_flags={}):
         super().__init__(teams, bracket, room_rank, flags, team_flags)
-        assert len(self.teams) == 4, "There must be four teams in a BPPairing"
```

Comment:
Not worth checking that `len(self.teams) == num_sides`?

Reply:
No, as the number of sides may actually vary.